### PR TITLE
implement accessibility

### PR DESCRIPTION
**Pull request by morethanreal**
add a new `modal` attribute, aria propertie, keep focus inside
a modal dialog.

@cdata PTAL cc @notwaldorf

### CODE REVIEW

File: demo/index.html [hunk 1]
```diff
@@ -37,9 +37,9 @@
       <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>
     </simple-dialog>
 
-    <button data-dialog="alert">alert</button>
+    <button data-dialog="alert">modal alert</button>
 
-    <simple-dialog id="alert" with-backdrop no-cancel-on-outside-click no-cancel-on-esc-key role="alertdialog">
```
Comment (cdata) [inline demo/index.html]: Hehe, those were some verbose attributes :scroll:

File: paper-dialog-behavior.html [hunk 2]
```diff
@@ -59,14 +64,101 @@
...
 
-    _updatClosingReasonConfirmed: function(confirmed) {
+    _updateAriaLabelledBy: function() {
+      var header = Polymer.dom(this).querySelector('h2');
+      if (!header) {
+        this.removeAttribute('aria-describedBy');
+        return;
+      }
+      var headerId = header.getAttribute('id');
+      if (headerId && this.getAttribute('aria-labelledBy') === headerId) {
```
Comment (cdata) [inline paper-dialog-behavior.html]: I don't think it matters, but `aria-labelledBy` doesn't need (or benefit from) camel case.

File: paper-dialog-behavior.html [hunk 3]
```diff
@@ -75,16 +167,58 @@
...
+              this._focusNode.focus();
+            }
+          } else {
+            this._lastFocusedElement = event.target;
+          }
+        }
+      }
+    },
+
+    _onBackdropClick: function() {
```
Comment (cdata) [inline paper-dialog-behavior.html]: Nice :gem:

**Comment by cdata** [comment]
This might be useful some day: https://github.com/cdata/focus-observer

This looks really awesome :beers: LGTM

File: paper-dialog-behavior.html [hunk 4]
```diff
@@ -59,14 +64,101 @@
...
+      observer.observe(node, {
+        childList: true,
+        subtree: true
+      });
+      return observer;
+    },
+
+    _modalChanged: function() {
+      if (this.modal) {
+        this.setAttribute('aria-modal', 'true');
```
Comment (notwaldorf) [inline paper-dialog-behavior.html]: Maybe combine into `this.setAttribute('aria-modal', this.modal ? 'true' : 'false'` ?

**Comment by notwaldorf** [comment]
This is super cool!

File: demo/index.html (hunk 1 above)
Comment (morethanreal) [inline demo/index.html]: Long live `modal`!

File: paper-dialog-behavior.html (hunk 2 above)
Comment (morethanreal) [inline paper-dialog-behavior.html]: Fixed. Typo...

**Pull request merged**